src/features/session_replay/aggregate/index.js

```diff
@@ -0,0 +1,304 @@
+      raw: true,
+      gzip: true
+    }, this)
+
+    // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
+    // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
+    registerHandler('errorAgg', (e) => {
+      this.hasError = true
+      // run once
+      // if the error was noticed AFTER the recorder was imported....
+      if (this.mode === MODE.ERROR) {
+        this.mode = MODE.full
+        if (recorder) {
+          this.scheduler.runHarvest()
+          this.stopRecording()
+          this.mode = MODE.FULL
+          this.startRecording()
+          this.scheduler.startTimer(this.harvestTimeSeconds)
+
+          const { session } = getRuntime(this.agentIdentifier)
+          session.state.sessionReplay = this.mode
+        }
+      }
+    }, this.featureName, this.ee)
+
+    // new handler for waiting for multiple flags.  will be useful if/when backend designs multiple flags, or for evaluating multiple feature flags simultaneously (stn vs sr)
+    this.waitForFlags(['sr']).then(([{ value }]) => {
+      this.initializeRecording(
+        value,
+        Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.errorSampleRate'),
+        Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.sampleRate')
+      )
+    })
+
+    drain(this.agentIdentifier, this.featureName)
+  }
+
+  /**
+   * Evaluate entitlements and sampling before starting feature mechanics, importing and configuring recording library, and setting storage state
+   * @param {boolean} entitlements - the true/false state of the "sr" flag from RUM response
+   * @param {boolean} errorSample - the true/false state of the error sampling decision
+   * @param {boolean} fullSample - the true/false state of the full sampling decision
+   * @returns {void}
+   */
+  async initializeRecording (entitlements, errorSample, fullSample) {
+    this.initialized = true
+    if (!entitlements) return
+
+    const { session } = getRuntime(this.agentIdentifier)
+    // if theres an existing session replay in progress, there's no need to sample, just check the entitlements response
+    // if not, these sample flags need to be checked
+    // if this isnt the FIRST load of a session AND
+    // we are not actively recording SR... DO NOT import or run the recording library
+    // session replay samples can only be decided on the first load of a session
+    // session replays can continue if already in progress
+    if (!session.isNew && session.state.sessionReplay === MODE.OFF) return
+
+    if (session.state.sessionReplay === MODE.FULL) this.mode = MODE.FULL
+    else if (session.isNew && fullSample) this.mode = MODE.FULL
+    else if (session.state.sessionReplay === MODE.ERROR) this.mode = MODE.ERROR
+    else if (errorSample) this.mode = MODE.ERROR
+    else return
```

Comment:
This is still really confusing. How about this?
```suggestion
    if (!session.isNew) { // inherit the mode of the existing session
        this.mode = session.state.sessionReplay
    } else { // determine the mode the new session should start in
        if (fullSample) this.mode = MODE.FULL // full mode has precedence over error mode
        else if (errorSample) this.mode = MODE.ERROR
        // If neither are selected, then the mode stays off.
    }
    if (this.mode === MODE.FULL) return // finally, if the mode is off, don't start rrweb
```